src/client/Client.cpp

```diff
@@ -304,14 +304,22 @@ void Client::mode(const Message &m) {
 }
 
 void Client::privmsg(const Message &m) {
-  Message reply(m);
-
-  reply.prefix = ident_->toString();
-  if (util::isChannelPrefix(reply.params[0][0])) {
-    ChannelMap &cm = server.getChannels();
+  if (m.params.empty() or m.params[0].empty()) {
+    send(Message::as_numeric_reply(util::ERR_NORECIPIENT, VA((m.command))));
+    return;
+  }
+  if (m.params.size() < 2 or m.params[1].empty()) {
+    send(Message::as_numeric_reply(util::ERR_NOTEXTTOSEND,
+                                   VA(("No text to send"))));
+    return;
+  }
+  const Message reply = {ident_->toString(), m.command, m.params};
+  const std::string recipient = reply.params[0];
```

Comment:
별거 아닌 부분이긴 한데 이 부분은 참조형이 더 좋을 것 같아요
```suggestion
  const std::string &recipient = reply.params[0];
```